### DISCUSSION TITLE
Content Hash in webpack config

### DISCUSSION THREAD

**Discussion by deepanshk** [webpack/webpack#17966] [Q&A]
I have angular js project, Trying to setup webpack as build process.
I have index.html which have some script tags like`<script type="text/javascript" src="main.js"></script>`, after creating build, it is changing to `<script type="text/javascript" src="bb03e28771c5d073f1a0.js"></script>`. I don't want to change any of file name.

Here is webpack config.

`const path = require("path");
const HtmlWebpackPlugin = require("html-webpack-plugin");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const TerserPlugin = require("terser-webpack-plugin");
const webpack = require("webpack");
const { CleanWebpackPlugin } = require("clean-webpack-plugin");
const glob = require("glob");
const ESLintPlugin = require("eslint-webpack-plugin");

const MergeIntoSingleFilePlugin = require("webpack-merge-and-include-globally");
var HtmlWebpackTagsPlugin = require("html-webpack-tags-plugin");

var angularJs = [
    "app/lib/typedarray.js",
    "app/bower_components/jquery/dist/jquery.min.js",
    "app/bower_components/bootstrap/dist/js/bootstrap.js",
    "app/bower_components/angular/angular.min.js",
    "app/bower_components/angular-route/angular-route.min.js",
    "app/lib/angular-upload.js",
    "app/bower_components/angular-animate/angular-animate.js",
    "app/bower_components/angular-ui-router/release/angular-ui-router.js",
    "app/bower_components/angular-ui-calendar/src/calendar.js",
    "app/bower_components/angular-google-chart/ng-google-chart.min.js",
    "app/bower_components/angular-ui-select/dist/select.min.js",
    "app/bower_components/angular-filter/dist/angular-filter.min.js",
];

var jsLibFiles = [
    "app/bower_components/lodash/dist/lodash.core.min.js",
    "app/bower_components/lodash/dist/lodash.min.js",
    "app/bower_components/angular-bootstrap/ui-bootstrap-tpls.min.js",
    "app/bower_components/moment/min/moment.min.js",
    "app/bower_components/moment-timezone/builds/moment-timezone-with-data.js",
    "app/bower_components/jquery-bridget/jquery-bridget.js",
    "app/bower_components/ev-emitter/ev-emitter.js",
    "app/bower_components/desandro-matches-selector/matches-selector.js",
    "app/bower_components/fizzy-ui-utils/utils.js",
    "app/bower_components/get-size/get-size.js",
    "app/bower_components/outlayer/item.js",
    "app/bower_components/outlayer/outlayer.js",
    "app/bower_components/masonry/masonry.js",
    "app/bower_components/imagesloaded/imagesloaded.js",
    "app/bower_components/angular-masonry/angular-masonry.js",
    "app/bower_components/fullcalendar/dist/fullcalendar.min.js",
    "app/bower_components/ngprogress/build/ngprogress.min.js",
    "app/bower_components/sweetalert/docs/assets/sweetalert/sweetalert.min.js",
    "app/bower_components/nprogress/nprogress.js",
    "app/bower_components/bootstrap-switch/dist/js/bootstrap-switch.min.js",
    "app/bower_components/angular-bootstrap-switch/dist/angular-bootstrap-switch.min.js",
    "app/bower_components/angular-bootstrap-lightbox/dist/angular-bootstrap-lightbox.min.js",
    "app/lib/angular-drag-and-drop-lists.js",
    "app/lib/ie-shims.js",
    "app/lib/utilis.js",
    "app/bower_components/angular-toastr/dist/angular-toastr.min.js",
    "app/bower_components/angular-toastr/dist/angular-toastr.tpls.min.js",
    "app/bower_components/angularjs-slider/dist/rzslider.min.js",
    "app/bower_components/angular-ui-tree/dist/angular-ui-tree.js",
    "app/bower_components/textAngular/dist/textAngular-rangy.min.js",
    "app/bower_components/textAngular/dist/textAngular-sanitize.min.js",
    "app/bower_components/textAngular/dist/textAngular.min.js",
    "app/bower_components/jquery-slimscroll/jquery.slimscroll.min.js",
    "app/lib/socket.io.js",
    "app/lib/modernizr.js",
    "app/lib/angular-desktop-notification.js",
    "app/bower_components/angular-xeditable/dist/js/xeditable.min.js",
    "app/bower_components/angular-ui-mask/dist/mask.js",
    "app/bower_components/pluralize/pluralize.js",
    "app/bower_components/mark.js/dist/mark.min.js",
    "app/bower_components/radialIndicator/dist/radialIndicator.min.js",
    "app/bower_components/radialIndicator/angular.radialIndicator.js",
    "app/bower_components/angular-wizard/dist/angular-wizard.min.js",
    "app/lib/ng-browser-update.js",
    "app/bower_components/angular-ui-scroll/dist/ui-scroll.min.js",
    "app/bower_components/angular-ui-scroll/dist/ui-scroll-grid.min.js",
];

const angularLibs = angularJs.map((file) => path.resolve(__dirname, file));
const resolvedJsLibFiles = jsLibFiles.map((file) =>
    path.resolve(__dirname, file)
);

module.exports = {
    entry: {
        "app.min": [
            ...glob.sync(path.resolve(__dirname, "app/models/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/controllers/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/animations/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/services/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/factories/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/directives/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/filters/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/components/**/*.js")),
            ...glob.sync(
                path.resolve(__dirname, "hybrent-react/dist/**/app.bundle.js")
            ),
            ...glob.sync(path.resolve(__dirname, "app/framework/**/*.js")),
        ],
    },

    mode: "development",
    devtool: "inline-source-map",
    devServer: {
        static: {
            directory: path.join(__dirname, "app"), // Adjust the path to match your project structure.
        },
        hot: true,
        host: "localhost", // or '0.0.0.0' to allow access from other devices
        port: 8080,
    },
    module: {
        rules: [
            {
                test: /\.json$/,
                use: "json5-loader",
                include: [path.resolve(__dirname, "app")],
            },
            {
                test: /\.scss$/,
                use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"],
            },
            {
                test: /\.html$/,
                use: [
                    {
                        loader: "html-loader",
                        options: {
                            sources: {
                                scriptingEnabled: false,
                            },
                        },
                    },
                ],
            },
        ],
    },
    optimization: {
        splitChunks: {
            chunks: "all",
        },
        minimizer: [new TerserPlugin()],
        realContentHash: false
    },
    resolve: {
        extensions: [".js", ".html", ".json"],
        modules: [path.resolve(__dirname, "app")],
        fallback: {
            events: require.resolve("events/"),
        },
    },
    plugins: [
        new webpack.HotModuleReplacementPlugin(),
        new HtmlWebpackPlugin({
            template: "app/index.html",
            filename: "index.html",
            inject: "head",
            scriptLoading: "blocking",
            hash: false,
        }),
        new MergeIntoSingleFilePlugin(
            {
                files: {
                    "angularLib.js": [...angularLibs],
                    "lib.js": [...resolvedJsLibFiles],
                },
                hash: false,
            },
            (filesMap) => {
                console.log("generated files: ", filesMap); // eslint-disable-line no-console
            }
        ),
        new HtmlWebpackTagsPlugin({
            tags: ["angularLib.js", "lib.js"],
            append: false,
            hash: false,
        }),
        new MiniCssExtractPlugin({
            filename: "stylesheets/app.min.css",
        }),
        new CleanWebpackPlugin(),
        new ESLintPlugin({
            extensions: ["js"],
        }),
    ],
    output: {
        filename: "[name].js",
        path: path.resolve(__dirname, "build/dev"),
        clean: true,
    },
};
`

**Comment by alexander-akait** [comment]
Somewhere you have `[contenthash].js` (`output.filename` or `output.chunkFilename`), I don't see it in your provided configuration, please check it, maybe you have more configurations

**Reply by deepanshk** [reply]
@alexander-akait please refer below comment

**Comment by deepanshk** [comment]
```javascript
const path = require("path");
const HtmlWebpackPlugin = require("html-webpack-plugin");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const TerserPlugin = require("terser-webpack-plugin");
const webpack = require("webpack");
const { CleanWebpackPlugin } = require("clean-webpack-plugin");
const glob = require("glob");
const ESLintPlugin = require("eslint-webpack-plugin");

const MergeIntoSingleFilePlugin = require("webpack-merge-and-include-globally");
var HtmlWebpackTagsPlugin = require("html-webpack-tags-plugin");

var angularJs = [
    "app/lib/typedarray.js",
    "app/bower_components/jquery/dist/jquery.min.js",
    "app/bower_components/bootstrap/dist/js/bootstrap.js",
    "app/bower_components/angular/angular.min.js",
    "app/bower_components/angular-route/angular-route.min.js",
    "app/lib/angular-upload.js",
    "app/bower_components/angular-animate/angular-animate.js",
    "app/bower_components/angular-ui-router/release/angular-ui-router.js",
    "app/bower_components/angular-ui-calendar/src/calendar.js",
    "app/bower_components/angular-google-chart/ng-google-chart.min.js",
    "app/bower_components/angular-ui-select/dist/select.min.js",
    "app/bower_components/angular-filter/dist/angular-filter.min.js",
];

var jsLibFiles = [
    "app/bower_components/lodash/dist/lodash.core.min.js",
    "app/bower_components/lodash/dist/lodash.min.js",
    "app/bower_components/angular-bootstrap/ui-bootstrap-tpls.min.js",
    "app/bower_components/moment/min/moment.min.js",
    "app/bower_components/moment-timezone/builds/moment-timezone-with-data.js",
    "app/bower_components/jquery-bridget/jquery-bridget.js",
    "app/bower_components/ev-emitter/ev-emitter.js",
    "app/bower_components/desandro-matches-selector/matches-selector.js",
    "app/bower_components/fizzy-ui-utils/utils.js",
    "app/bower_components/get-size/get-size.js",
    "app/bower_components/outlayer/item.js",
    "app/bower_components/outlayer/outlayer.js",
    "app/bower_components/masonry/masonry.js",
    "app/bower_components/imagesloaded/imagesloaded.js",
    "app/bower_components/angular-masonry/angular-masonry.js",
    "app/bower_components/fullcalendar/dist/fullcalendar.min.js",
    "app/bower_components/ngprogress/build/ngprogress.min.js",
    "app/bower_components/sweetalert/docs/assets/sweetalert/sweetalert.min.js",
    "app/bower_components/nprogress/nprogress.js",
    "app/bower_components/bootstrap-switch/dist/js/bootstrap-switch.min.js",
    "app/bower_components/angular-bootstrap-switch/dist/angular-bootstrap-switch.min.js",
    "app/bower_components/angular-bootstrap-lightbox/dist/angular-bootstrap-lightbox.min.js",
    "app/lib/angular-drag-and-drop-lists.js",
    "app/lib/ie-shims.js",
    "app/lib/utilis.js",
    "app/bower_components/angular-toastr/dist/angular-toastr.min.js",
    "app/bower_components/angular-toastr/dist/angular-toastr.tpls.min.js",
    "app/bower_components/angularjs-slider/dist/rzslider.min.js",
    "app/bower_components/angular-ui-tree/dist/angular-ui-tree.js",
    "app/bower_components/textAngular/dist/textAngular-rangy.min.js",
    "app/bower_components/textAngular/dist/textAngular-sanitize.min.js",
    "app/bower_components/textAngular/dist/textAngular.min.js",
    "app/bower_components/jquery-slimscroll/jquery.slimscroll.min.js",
    "app/lib/socket.io.js",
    "app/lib/modernizr.js",
    "app/lib/angular-desktop-notification.js",
    "app/bower_components/angular-xeditable/dist/js/xeditable.min.js",
    "app/bower_components/angular-ui-mask/dist/mask.js",
    "app/bower_components/pluralize/pluralize.js",
    "app/bower_components/mark.js/dist/mark.min.js",
    "app/bower_components/radialIndicator/dist/radialIndicator.min.js",
    "app/bower_components/radialIndicator/angular.radialIndicator.js",
    "app/bower_components/angular-wizard/dist/angular-wizard.min.js",
    "app/lib/ng-browser-update.js",
    "app/bower_components/angular-ui-scroll/dist/ui-scroll.min.js",
    "app/bower_components/angular-ui-scroll/dist/ui-scroll-grid.min.js",
];

const angularLibs = angularJs.map((file) => path.resolve(__dirname, file));
const resolvedJsLibFiles = jsLibFiles.map((file) =>
    path.resolve(__dirname, file)
);


module.exports = {
    entry: {
        "app.min": [
            ...glob.sync(path.resolve(__dirname, "app/models/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/controllers/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/animations/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/services/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/factories/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/directives/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/filters/**/*.js")),
            ...glob.sync(path.resolve(__dirname, "app/components/**/*.js")),
            ...glob.sync(
                path.resolve(__dirname, "hybrent-react/dist/**/app.bundle.js")
            ),
            ...glob.sync(path.resolve(__dirname, "app/framework/**/*.js")),
        ],
    },

    mode: "development",
    devtool: "inline-source-map",
    devServer: {
        static: {
            directory: path.join(__dirname, "app"), // Adjust the path to match your project structure.
        },
        hot: true,
        host: "localhost", // or '0.0.0.0' to allow access from other devices
        port: 8080,
    },
    module: {
        rules: [
            {
                test: /\.json$/,
                use: "json5-loader",
                include: [path.resolve(__dirname, "app")],
            },
            {
                test: /\.scss$/,
                use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"],
            },
            {
                test: /\.html$/,
                use: [
                    {
                        loader: "html-loader",
                        options: {
                            sources: {
                                scriptingEnabled: true,
                            },
                        },
                    },
                ],
            },
        ],
    },
    optimization: {
        splitChunks: {
            chunks: "all",
        },
        minimizer: [new TerserPlugin()],
        realContentHash: false
    },
    resolve: {
        extensions: [".js", ".html", ".json"],
        modules: [path.resolve(__dirname, "app")],
        fallback: {
            events: require.resolve("events/"),
        },
    },
    plugins: [
        new webpack.HotModuleReplacementPlugin(),
        new HtmlWebpackPlugin({
            template: "app/index.html",
            filename: "index.html",
            inject: "head",
            scriptLoading: "blocking",
            hash: false,
        }),
        new MergeIntoSingleFilePlugin(
            {
                files: {
                    "angularLib.js": [...angularLibs],
                    "lib.js": [...resolvedJsLibFiles],
                },
                hash: false,
            },
            (filesMap) => {
                console.log("generated files: ", filesMap); // eslint-disable-line no-console
            }
        ),
        new HtmlWebpackTagsPlugin({
            tags: ["angularLib.js", "lib.js"],
            append: false,
            hash: false,
        }),
        new MiniCssExtractPlugin({
            filename: "stylesheets/app.min.css",
        }),
        new CleanWebpackPlugin(),
        new ESLintPlugin({
            extensions: ["js"],
        }),
    ],
    output: {
        filename: "[name].js",
        path: path.resolve(__dirname, "build/dev"),
        clean: true,
    },
};

```

**Reply by alexander-akait** [reply]
I don't see any `[contenthash]` in your configuration, how you run webpack?
